### PR TITLE
flatten: ignore falsy non-0 css values

### PR DESCRIPTION
**Pull request by wmertens**
This implements #1110.

Note that the output still includes spurious `;`, I have no idea where they come from, I even searched the code for all `;` :)

### CODE REVIEW

File: src/utils/flatten.js [hunk 1]
```diff
@@ -5,13 +5,18 @@ import isPlainObject from 'is-plain-object'
 import type { Interpolation } from '../types'
 
 export const objToCss = (obj: Object, prevKey?: string): string => {
-  const css = Object.keys(obj).map(key => {
-    if (isPlainObject(obj[key])) return objToCss(obj[key], key)
-    return `${hyphenate(key)}: ${obj[key]};`
-  }).join(' ')
-  return prevKey ? `${prevKey} {
+  const css = Object.keys(obj)
+    .filter(key => obj[key] === 0 || obj[key])
```
Comment (philpl) [inline src/utils/flatten.js]: Can you use the same check as the one we’re using inside flatten? It’d be nice to share it even if it shouldn’t make a difference here :)

Comment (philpl) [reply src/utils/flatten.js]: @wmertens Bump :wink:

**Comment by wmertens** [comment]
Oops missed that :)

I gave it a quick shot, first factoring out the test to `isInvalidCssValue`
and using it in both places, but then I couldn't commit because the flow
eslint didn't see that the function is filtering out falsy chunks, so now I
just copied the test verbatim. I must say I prefer my `obj[key] === 0 ||
obj[key]` but ok.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ All notable changes to this project will be documented in this file. If a contri
 - Fixed nonce missing from global styles (see [#1088](https://github.com/styled-components/styled-components/pull/1088))
 - Improve component mount and unmount performance with changes to `createBroadcast`. Deprecates usage of `CHANNEL` as a function, will be update to `CHANNEL_NEXT`'s propType in a future version. (see [#1048](https://github.com/styled-components/styled-components/pull/1048))
 - Fixed comments in react-native (see [#1041](https://github.com/styled-components/styled-components/pull/1041))
+- Don't emit empty rules like `color: ;` when using objects for CSS. Note that this still happens with template interpolations.
```
Comment (philpl) [inline CHANGELOG.md]: Can you refer to this PR in the changelog here please? :)

File: src/utils/flatten.js [hunk 3]
```diff
@@ -5,13 +5,26 @@ import isPlainObject from 'is-plain-object'
...
 export const objToCss = (obj: Object, prevKey?: string): string => {
-  const css = Object.keys(obj).map(key => {
-    if (isPlainObject(obj[key])) return objToCss(obj[key], key)
-    return `${hyphenate(key)}: ${obj[key]};`
-  }).join(' ')
-  return prevKey ? `${prevKey} {
+  const css = Object.keys(obj)
+    .filter(key => {
+      const chunk = obj[key]
+      return !(
```
Comment (philpl) [inline src/utils/flatten.js]: Can you instead just negate the reference equality? It's a lot easier to miss a negation up here.

Comment (wmertens) [reply src/utils/flatten.js]: Good point! Done.

**Comment by mxstbr** [comment]
Ah, I wanted to release this with `2.2.0`, my bad! Mind resolving the CHANGELOG conflict @wmertens? I'll get this out as 2.2.1!

**Comment by philpl** [comment]
@mxstbr We can land a patch together with https://github.com/styled-components/styled-components/pull/1069 potentially?

**Comment by wmertens** [comment]
Rebased :)

**Comment by mxstbr** [comment]
Awesome, thank you Wout!

**Pull request merged**